packages/lab/src/list-next/useList.ts

```diff
@@ -0,0 +1,284 @@
+  };
+
+  const updateScroll = (currentTarget: Element) => {
+    if (!list || !currentTarget) return;
+    const { offsetTop, offsetHeight } = currentTarget;
+    const listHeight = list?.clientHeight;
+    const listScrollTop = list?.scrollTop;
+    if (offsetTop < listScrollTop) {
+      list.scrollTop = offsetTop;
+    } else if (offsetTop + offsetHeight > listScrollTop + listHeight) {
+      list.scrollTop = offsetTop + offsetHeight - listHeight;
+    }
+  };
+
+  // Handlers
+  const handleClick = ({ currentTarget }: MouseEvent<HTMLUListElement>) => {
+    const nonClickableTarget = activeOptions.indexOf(currentTarget) === -1;
+    if (nonClickableTarget) {
+      return;
+    }
+    toggleSelectItem(currentTarget);
+    setActiveDescendant(currentTarget.id);
+    setFocusedIndex(null);
+    updateScroll(currentTarget);
+  };
+
+  const handleBlur = useEventCallback(() => {
+    setFocusedIndex(null);
+    if (onBlur) {
+      onBlur();
+    }
+  });
+
+  const handleMouseDown = useEventCallback(() => {
+    setMouseDown(true);
+    onMouseDown()
+  });
+
+  const handleFocus = useEventCallback(() => {
+    if (!activeDescendant && !mouseDown) {
+      // Focus on first active option if no option was previously focused
+      focusFirstItem();
+    } else {
+      const activeElement = document.getElementById(activeDescendant);
+      if (activeElement) {
+        justFocusItem(activeElement);
+      }
+    }
+  });
+
+  const handleKeyDown = useEventCallback((evt: KeyboardEvent) => {
+    const { key } = evt;
+    const currentItem =
+      document.getElementById(activeDescendant) || activeOptions[0];
+    let nextItem = currentItem;
+    if (!currentItem) {
+      return;
+    }
+    switch (key) {
+      case PageUp:
+      case PageDown:
+        nextItem =
+          key === PageUp
+            ? findPreviousOption(currentItem, displayedItemCount)
+            : findNextOption(currentItem, displayedItemCount);
+
+        if (nextItem && nextItem !== currentItem) {
+          evt.preventDefault();
+          justFocusItem(nextItem);
+        }
+        break;
+      case ArrowUp:
+      case ArrowDown:
+        nextItem =
+          key === ArrowUp
+            ? findPreviousOption(currentItem, 1)
+            : findNextOption(currentItem, 1);
+
+        if (nextItem && nextItem !== currentItem) {
+          evt.preventDefault();
+          justFocusItem(nextItem);
+        }
+        break;
+      case Home:
+        evt.preventDefault();
+        focusFirstItem();
+        break;
+      case End:
+        evt.preventDefault();
+        focusLastItem();
+        break;
+      case Space:
+        evt.preventDefault();
+        toggleSelectItem(nextItem);
+        break;
+      default:
+        break;
+    }
+    if (onKeyDown) {
+      onKeyDown(currentItem);
+    }
+    return;
+  });
+
+  // Effects
+  useEffect(() => {
+    // sets list in first render
+    list = listRef.current;
+  }, []);
+
+  useEffect(() => {
```

Comment:
event listeners and available options state are set in this effect, that updates everytime the list ref changes